### PR TITLE
Tsunkaraneni/create ui definition

### PR DESCRIPTION
**Pull request by TSunny007**
_The only change I have made is adding a UiDefinition field to the TemplateSpecsVersion object._

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by TarunSunkaraneni** [comment]
@snarkywolverine

File: specification/resources/resource-manager/Microsoft.Resources/preview/2020-10-01-preview/templateSpecs.json [hunk 1]
```diff
@@ -0,0 +1,867 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "TemplateSpecsClient",
+    "description": "The APIs listed in this specification can be used to manage Template Spec resources through the Azure Resource Manager.",
+    "version": "2019-06-01-preview"
```
Comment (majastrz) [inline specification/resources/resource-manager/Microsoft.Resources/preview/2020-10-01-preview/templateSpecs.json:6]: >2019-06-01-preview [](start = 16, length = 18)

You have the wrong API version in this file and in all the examples.

**Review by majastrz — CHANGES_REQUESTED** [review]
The API addition looks good, but you need to fix the checks.

**Review by majastrz — APPROVED** [review]
Signed off from ARM side.

**Comment by majastrz** [comment]
@allenjzhang THere are 3 "automation - sdk" CI jobs failing on this PR. I suspect it's some infra issue unrelated to @TarunSunkaraneni's changes. Can you confirm?

**Comment by allenjzhang** [comment]
> @allenjzhang THere are 3 "automation - sdk" CI jobs failing on this PR. I suspect it's some infra issue unrelated to @TarunSunkaraneni's changes. Can you confirm?

Re-running it now. Waiting.

**Review by allenjzhang — APPROVED** [review]
Compared with previous version, single change on UIDefinition. Looks good

**Pull request merged**